File: 66-plus-one/66-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int newDigits[] = new int[digits.length + 1];
        int carry = 1;
        for (int i = digits.length - 1; i >= 0; i--) {
            if (digits[i] + carry < 10) {
                digits[i] = digits[i] + carry;
                break;
            } else {
                newDigits[i + 1] = 0;
                digits[i] = 0;
            }
        }

        if (digits[0] == 0) {
            newDigits[0] = 1;
            return newDigits;
        }
        return digits;
    }
}

File: 66-plus-one/66-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry=0;
        reverse(digits.begin(),digits.end());
        digits[0]=digits[0]+1;
        carry=digits[0]/10;
        digits[0]=digits[0]%10;
        for(int i=1;i<digits.size();i++)
        {
            digits[i]=digits[i]+carry;
            carry=digits[i]/10;
            digits[i]=digits[i]%10;
        }
        if(carry==1)
            digits.push_back(carry);
        reverse(digits.begin(),digits.end());
        return digits;
    }
};